### PR TITLE
Add check for nil in parameter sanitization

### PR DESCRIPTION
**Pull request by devonestes**
I will start by appologizing for what will surely be a
not-super-well-done job of investigating a bug that I was having.

So, for some reason today, a process that had been working stopped
working. When I attempted to call `.save` or `.save!` on a model, I got
an error `No method 'to_sym` for NilClass`, even though I'm just trying
to do a very simple update of a very simple model (no Attributes API
stuff, or even any aggregate attributes to speak of).

All I know, is that somehow in the attributes being passed into
this method, a key of `nil` has snuck in there somehow, causing the
error when we try and call `to_sym` on that. My solution was to just
skip that key-value pair if the attribute is `nil`. I figured this
would be a good thing to do anyway.

So, long story short, I don't know how this is possible or why it's
happening, but I know this patch fixes my code in production, and it
seems like a reasonable defensive coding thing to do. Maybe it'll help
others as well!

### CODE REVIEW

**Comment by devonestes** [comment]
The CI failure appears to be unrelated to this change (it appears to be an IO error): https://travis-ci.org/rails/rails/jobs/219004419

I'd be happy to close and re-open this to re-run CI, though, if that's helpful.

**Comment by matthewd** [comment]
Going out of our way to swallow invalid data doesn't seem like a good idea.

For that matter, if you don't know where the `nil` key is coming from, is it really safe to assume its value can be summarily ignored?

Thanks for the PR, but I don't think this is the right way for us to go.
